presto-main/src/test/java/io/prestosql/sql/planner/TestDynamicFilter.java

```diff
@@ -259,6 +261,33 @@ public void testNestedDynamicFiltersRemoval()
                                                         tableScan("orders", ImmutableMap.of("ORDERS_CK27", "clerk"))))), metadata)));
     }
 
+    @Test
+    public void testNonPushedDownJoinFilterRemoval()
+    {
+        PlanMatchPattern t0 = project(node(FilterNode.class,
+                tableScan("part", ImmutableMap.of("K0", "partkey", "V0", "size"))));
+        PlanMatchPattern t1 = exchange(project(node(FilterNode.class,
+                tableScan("part", ImmutableMap.of("K1", "partkey", "V1", "size")))));
+        PlanMatchPattern t2 = exchange(project(
+                tableScan("part", ImmutableMap.of("K2", "partkey", "V2", "size"))));
+
+        PlanMatchPattern join1 = join(
+                INNER,
+                ImmutableList.of(equiJoinClause("K0", "K1")),
+                t0, t1);
+        PlanMatchPattern project1 = project(project(ImmutableMap.of("S", expression("V0 + V1")), join1));
+        PlanMatchPattern join2 = join(
+                INNER,
+                ImmutableList.of(equiJoinClause("K0", "K2"), equiJoinClause("S", "V2")),
+                project1, t2);
+        PlanMatchPattern matcher = anyTree(project(join2));
+
+        assertPlan("SELECT 1 FROM part t0, part t1, part t2 " +
+                        "WHERE t0.partkey = t1.partkey AND t0.partkey = t2.partkey " +
+                        "AND t0.size + t1.size = t2.size",
+                noJoinReordering(), matcher);
```

Comment:
Could you inline `PlanMatchPattern` here ? It's easier to see the plan shape when there is only one pattern to look at, similar to the other test cases

Reply:
You're right, fixed.

Reply:
You're right, fixed.